pkg/virt-controller/watch/vm/vm.go

```diff
@@ -3262,3 +3274,149 @@ func hasEphemeralHotplugVolumes(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMa
 
 	return false
 }
+
+func (c *Controller) patchHotplugVolumes(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance) error {
+	if !vmi.IsRunning() {
+		return nil
+	}
+
+	newVmiVolumes := append(filterHotplugVMIVolumes(vm, vmi), getNewHotplugVMVolumes(vm, vmi)...)
+	newVmiDisks := append(filterHotplugVMIDisks(vm, vmi, newVmiVolumes), getNewHotplugVMDisks(vm, vmi, newVmiVolumes)...)
+
+	if equality.Semantic.DeepEqual(vmi.Spec.Volumes, newVmiVolumes) &&
+		equality.Semantic.DeepEqual(vmi.Spec.Domain.Devices.Disks, newVmiDisks) {
+		log.Log.Object(vm).V(3).Info("No hotplug volumes to patch")
+		return nil
+	}
+
+	patchSet := patch.New(
+		patch.WithTest("/spec/volumes", vmi.Spec.Volumes),
+		patch.WithTest("/spec/domain/devices/disks", vmi.Spec.Domain.Devices.Disks),
+	)
```

Comment:
do you need the test option if the next operation is an "add"?

Reply:
Will fail either way but I think the error is better if the test is there. What's the downside of having it?

Reply:
Yeah, there are fields in containerdisk but not PersistentVolumeclaim/DataVolume. See https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/vm/vm.go#L815-L825

Reply:
Yeah, I hope I didn't forget anything :D

Reply:
But you are right, that loop should be enough. I haven't see anything for PVCs or DVs which was missing for the other update strategy.